app/services/match_assignment.rb

```diff
@@ -0,0 +1,57 @@
+class MatchAssignment
+  def initialize(user)
+    @user = user
+    @success = false
+  end
+
+  def perform
+    Match.transaction do
```

Comment:
Why the double transaction? Can't you wrap in `ActiveRecord::Base.transaction` only?

Reply:
Oops, this should've been a single on. My bad.